### PR TITLE
[ADD] duplicate-po-message-definition, po-syntax-error, po-msgstr-variables: Enable PO checks

### PR DESCRIPTION
**Pull request by moylop260**
* duplicate-po-message-definition
  - Check duplicated message-id in po files

* po-syntax-error
  - Check syntax error in po files

* po-msgstr-variables
  - Check different number of string variables translated (e.g. %s or {})

Closes https://github.com/OCA/pylint-odoo/issues/289 https://github.com/OCA/pylint-odoo/issues/274

cc @luisg123v @randall-vx 

UPDATED: 
```bash
cd pylint-odoo  # root path of this repository
pip install -r requirements.txt
pylint -d all -e po-syntax-error,duplicate-po-message-definition,po-msgstr-variables --load-plugins=pylint_odoo {YOUR-BROKEN-MODULE}
```

### CODE REVIEW

File: pylint_odoo/checkers/modules_odoo.py [hunk 1]
```diff
@@ -439,6 +451,58 @@ def visit_tryexcept(self, node):
...
                 self.add_message('except-pass', node=handler)
 
+    def _get_po_line_number(self, po_entry):
+        """Get line number of a PO entry similar to 'msgfmt' output
+        entry.linenum returns line number of the definition of the entry
+        'msgfmt' returns line number of the 'msgid'
+        This method get line number of the 'msgid'
+        """
+        linenum = po_entry.linenum
+        for line in po_entry.__str__().split('\n'):
```
[the remark this answers is not in the document]
Comment (moylop260) [reply pylint_odoo/checkers/modules_odoo.py:461]: What do you mean?

File: pylint_odoo/checkers/modules_odoo.py [hunk 2]
```diff
@@ -439,6 +451,58 @@ def visit_tryexcept(self, node):
...
+                fname = os.path.join(self.module_path, po_file)
+                msg = str(oe).replace(fname + ' ', '').strip()
+                self.msg_args.append((po_file, msg))
+
+    def _check_duplicate_po_message_definition(self):
+        """Check duplicate message definition (message-id)
+        in all entries of PO files
+        """
+        self.msg_args = []
+        for po_file in self.filter_files_ext('po'):
```
[the remark this answers is not in the document]
Comment (moylop260) [reply pylint_odoo/checkers/modules_odoo.py:483]: Good point!
Thank you

File: pylint_odoo/checkers/modules_odoo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (luisg123v) [reply pylint_odoo/checkers/modules_odoo.py]: I mean:

`for line in str(po_entry).split('\n'):`

**Comment by moylop260** [comment]
@luisg123v 

> Remember to run the command to update the README.

It will be re-build in the next release version as usual.


> There is another case that it's not considered here, I don't know if it could be also considered, which is the same term translated several times:

FYI running the same `msgfmt` command of the original issue:
 - https://github.com/OCA/pylint-odoo/issues/289

The output is empty for this case of use:
 - image

But using same message-id:
 - image

Maybe It should be a different check.

**Comment by luisg123v** [comment]
@moylop260 yes, that case unfortunately is not detected as a duplicate, because it's not a `.po` error, but an Odoo error.

**Comment by moylop260** [comment]
@luisg123v 
IMHO It is a different check since that is not a `duplicate-po-message-definition` really but `duplicate-po-msgid-odoo` (or something like that)

Could you create a new issue, please?
Add a real example module failing in Odoo but not `msgfmt`

> Note:
It doesn't apply for views, as views may contain several terms, but they're referred using the same header.

Consider adding both cases a valid case where Odoo doesn't fail and another one failing.

**Comment by luisg123v** [comment]
@moylop260 I like the idea.

File: pylint_odoo/checkers/modules_odoo.py [hunk 3]
```diff
@@ -439,6 +456,84 @@ def visit_tryexcept(self, node):
...
+            try:
+                po = polib.pofile(os.path.join(self.module_path, po_file),
+                                  encoding='UTF-8')
+            except (IOError, OSError):
+                # If there is an error it is another check
+                continue
+            for entry in po:
+                if not entry.msgstr:
+                    # skip unstralated entry
```
Comment (luisg123v) [inline pylint_odoo/checkers/modules_odoo.py:524]:
```suggestion
                    # skip untranslated entry
```

**Comment by moylop260** [comment]
- I tested locally using all my local repositories and it didn't raise an error and the checks detected are valid.
 - It has corner cases valid and invalid, found before, in the unittest
 - New methods have 100% of coverage
 - I tested running in odoo/odoo addons and it detected valid cases
 - I tested it using py2 and py3

All looks working as expect
Thanks for feedback

**Comment by randall-vx** [comment]
Tested in customer main app. Good job Moy !

**Pull request merged**